pkg/asset/installconfig/aws/validation.go

```diff
@@ -298,8 +300,17 @@ func validateServiceEndpoints(fldPath *field.Path, region string, services []aws
 		}
 		conn.Close()
 	}
-	if _, partitionFound := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region); partitionFound {
-		return allErrs
+
+	if partition, partitionFound := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region); partitionFound {
+		if _, ok := partition.Regions()[region]; ok {
+			return allErrs
+		} else if len(services) == 0 {
+			err := validateRegion(region)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("region"), region, "region not found"))
```

Comment:
Let's include the underlying error in the field error.
```suggestion
				allErrs = append(allErrs, field.Invalid(fldPath.Child("region"), region, err.Error()))
```